keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1128 @@
+logs or events for this purpose.
+-->
+
+Yes if `recursiveReadOnly` is present in a Pod manifest.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [X] Events
+  - Event Reason: pod creation
+<!--
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [ ] Other (treat as last resort)
+  - Details:
+-->
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+No CrashLoopBackOff.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [X] Metrics
+  - Metric name: CrashloopBackOff
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+<!--
+Think about both cluster-level services (e.g. metrics-server) as well
+as node-level agents (e.g. specific version of CRI). Focus on external or
+optional services that are needed. For example, if this feature depends on
+a cloud provider API, or upon an external software-defined storage or network
+control plane.
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+No
+
+### Scalability
```

Comment:
Not answered

Reply:
Added
```
Each of the nodes will have indepedent status about the availability of RRO,
so this proposal does not have scalability issues.
```

Reply:
Do we have a mechanism to expose which CRI `RuntimeHandlerFeatures` a given node supports?

Reply:
> Do we have a mechanism to expose which CRI `RuntimeHandlerFeatures` a given node supports?

Currently just crictl.

Eventually `kubectl get runtimeclass` should expose the detailed information, but I guess it can be another KEP.

Reply:
I'd call this out somewhere - there is currently no way for users or higher-layer components to know if a node supports RRO except to try to run a pod.  Pods which fail because RRO=Enabled may get rescheduled onto nodes which can't run them.

We probably need to define a metric (or increment if it exists) saying how often kubelet rejects pods, and why.

Reply:
Added

```
### Scalability

<!--
For alpha, this section is encouraged: reviewers should consider these questions
and attempt to answer them.

For beta, this section is required: reviewers must answer these questions.

For GA, this section is required: approvers should be able to confirm the
previous answers based on experience in the field.
-->

A pod with `RRO=Enabled` may be rejected by kubelet with the probablility of $$B/A$$,
where $$A$$ is the number of all the nodes that may potentially accept the pod,
and $$B$$ is the number of the nodes that may potentially accept the pod but does not support RRO.
This may affect scalability.
```